src/iso/cloud_init_iso.cpp

```diff
@@ -714,16 +714,29 @@ void mp::CloudInitFileOps::update_cloud_init_with_new_extra_interfaces_and_new_i
     meta_data_file_content =
         mpu::emit_cloud_config(mpu::make_cloud_init_meta_config_with_id_tweak(meta_data_file_content, new_instance_id));
 
-    if (extra_interfaces.empty())
-    {
-        iso_file.erase("network-config");
```

Comment:
It is again a matter of the old and new format of the `network-config`, the new format always has the default interface, so it is always present. Then updating the cloud-init becomes taking the `default_mac_addr` and `extra_interfaces` and creating the string and writing to `iso_file["network-config"]`. This is also backward compatible because if it is the old format and the `iso_file["network-config"]` does not exist, it will be updated to the new format. 

The removed code is purely based on the old format, meaning `if (extra_interfaces.empty())` then, the `network-config` node should not exist. Hence, we have the first part `iso_file.erase`.

Reply:
Got it.